vamp-server/convert.cpp

```diff
@@ -556,8 +558,15 @@ readInput(string format, RequestOrResponse::Direction direction, bool &eof)
     if (format == "json") {
         string err;
         auto result = readInputJson(direction, err, eof);
-        if (err != "") throw runtime_error(err);
-        else return result;
+        const bool isRecognisedError = !result.success && result.errorText != "";
+        
+        // if the RequestOrResponse (result) has been populated with success=false and an error message
+        // then the server returned a well formed error, it is safe to return it for conversion 
+        // -- but if err is populated, something else has gone wrong
+        if (isRecognisedError || err == "") 
+            return result;
```

Comment:
Good point. Here I will use the excuse that there were not braces [before](https://github.com/piper-audio/piper-cpp/blob/master/vamp-server/convert.cpp#L559), but they were one-lined.